src/snowflake/snowpark/mock/_plan.py

```diff
@@ -589,25 +589,34 @@ def execute_mock_plan(
 
         from_df = execute_mock_plan(from_, expr_to_alias)
 
-        result_df = TableEmulator()
-
+        columns = []
+        data = []
+        sf_types = []
+        null_rows_idxs_map = {}
         for exp in projection:
             if isinstance(exp, Star):
-                for i in range(len(from_df.columns)):
-                    result_df.insert(len(result_df.columns), str(i), from_df.iloc[:, i])
-                result_df.columns = from_df.columns
-                result_df.sf_types = from_df.sf_types
-                result_df.sf_types_by_col_index = from_df.sf_types_by_col_index
+                data += [d for _, d in from_df.items()]
+                columns += list(from_df.columns)
+                sf_types += list(
+                    from_df.sf_types_by_col_index.values()
+                    if from_df.sf_types_by_col_index
+                    else from_df.sf_types.values()
+                )
+                null_rows_idxs_map.update(from_df._null_rows_idxs_map)
             elif (
                 isinstance(exp, UnresolvedAlias)
                 and exp.child
                 and isinstance(exp.child, Star)
             ):
                 for e in exp.child.expressions:
-                    col_name = analyzer.analyze(e, expr_to_alias)
-                    result_df[col_name] = calculate_expression(
+                    column_name = analyzer.analyze(e, expr_to_alias)
+                    columns.append(column_name)
+                    column_series = calculate_expression(
                         e, from_df, analyzer, expr_to_alias
                     )
+                    data.append(column_series)
+                    sf_types.append(column_series.sf_type)
+                    null_rows_idxs_map[column_name] = column_series._null_rows_idxs
```

Comment:
There are sql statements that will return tables with duplicate names so I think this is a valid scenario. Using those columns in future transformations almost always leads to an error though so I would not recommend this scenario.

Reply:
I see, I asked because I'm wondering if null_rows_idxs_map[column_name] could be overwritten by a following up column of the same name, which leads to some sort of data corruption.

so sounds like it is possible, should we raise error in this case?

Reply:
based upon our discussion, column of the same name implies the same data, so no need to worry about